### PR TITLE
modifications for OJS 3.1.2, fix role bug, application-dependent path

### PR DESCRIPTION
**Pull request by mpbraendle**

### CODE REVIEW

File: controllers/grid/CustomLocaleGridHandler.inc.php [hunk 1]
```diff
@@ -71,8 +72,11 @@ function updateLocale($args, $request) {
 			// save changes
 			$changes = $args['changes'];
 
+			/* UZH CHANGE OJS-72 2019/02/26/mb */
+			/* $customFilesDir = Config::getVar('files', 'public_files_dir') .
+				"/presses/$contextId/" . CUSTOM_LOCALE_DIR . "/$locale"; */
 			$customFilesDir = Config::getVar('files', 'public_files_dir') .
-				"/presses/$contextId/" . CUSTOM_LOCALE_DIR . "/$locale";
+                                "/journals/$contextId/" . CUSTOM_LOCALE_DIR . "/$locale";
```
Comment (asmecher) [inline controllers/grid/CustomLocaleGridHandler.inc.php]: This still refers to `journals`, rather than the app-agnostic directory name.

File: controllers/grid/CustomLocaleGridHandler.inc.php [hunk 2]
```diff
@@ -241,8 +245,10 @@ function index($args, $request) {
 	 */
 	function getFilterForm() {
 		$customLocalePlugin = self::$plugin;
-		$templatePath = $customLocalePlugin->getTemplatePath();
-		return $templatePath . 'customLocaleGridFilter.tpl';
+		/* UZH CHANGE OJS-74 2019/03/05/mb adapt to OJS 3.1.2 */
+		/* $templatePath = $customLocalePlugin->getTemplatePath(); */
+		/* return $templatePath . 'customLocaleGridFilter.tpl'; */
+		return $customLocalePlugin->getTemplateResource('customLocaleGridFilter.tpl');
```
Comment (asmecher) [inline controllers/grid/CustomLocaleGridHandler.inc.php]: I've split the template adaptations for the `stable-3_1_2` (just created) and `master` branches into a separate commit: https://github.com/asmecher/customLocale/commit/bca9b8a46099791360d54c18a59cf26331d25582

File: CustomLocalePlugin.inc.php [hunk 3]
```diff
@@ -20,40 +20,44 @@ class CustomLocalePlugin extends GenericPlugin {
...
-
-				import('lib.pkp.classes.file.FileManager');
-				$fileManager = new FileManager();
-				foreach ($localeFiles as $localeFile) {
-					$customLocalePath = $customLocalePathBase . $localeFile->getFilename();
-					if ($fileManager->fileExists($customLocalePath)) {
-						AppLocale::registerLocaleFile($locale, $customLocalePath, false);
-					}
+	/* UZH CHANGE OJS-74 2019/03/05/mb adapt to OJS 3.1.2 */
+	function register($category, $path, $mainContextId = null) {
```
Comment (asmecher) [inline CustomLocalePlugin.inc.php]: Moved the warning message resolution into a separate commit: https://github.com/asmecher/customLocale/commit/7a5cfcdf29447950c5fb5bab969937a16ce9cd5e

File: CustomLocaleHandler.inc.php [hunk 4]
```diff
@@ -22,7 +22,12 @@ function printCustomLocaleChanges($args, $request) {
 		$contextId = $context->getId();
 
 		$publicFilesDir = Config::getVar('files', 'public_files_dir');
-		$customLocaleDir = "$publicFilesDir/presses/$contextId/" . CUSTOM_LOCALE_DIR;
+		/* UZH CHANGE OJS-71 2019/03/22/mb */
+		/* $customLocaleDir = "$publicFilesDir/presses/$contextId/" . CUSTOM_LOCALE_DIR; */
+		$fileDirectories = Application::getFileDirectories();
```
Comment (asmecher) [inline CustomLocaleHandler.inc.php]: I would suggest using `PublicFileManager::getContextFilesPath` instead of `Application::getFileDirectories`.

**Comment by asmecher** [comment]
Thanks, @mpbraendle! I made a few suggestions and broke the changes out of a single commit where they weren't relevant to the main issue (the path fix). My other concern is about the path change is that it'll mean existing users of the plugin need to manually move contents -- this should be noted in the plugin documentation. (Plugins don't yet have a good toolset for update scripts; that's on my list.) Would you mind adding a note to the documentation?

File: controllers/grid/CustomLocaleGridHandler.inc.php (hunk 1 above)
Comment (asmecher) [reply controllers/grid/CustomLocaleGridHandler.inc.php]: (Also, watch out for space-based indents -- it looks like some have been added here instead of tabs.)

Comment (mpbraendle) [reply controllers/grid/CustomLocaleGridHandler.inc.php]: vi. Sigh ... . Will look out.

File: CustomLocaleHandler.inc.php (hunk 4 above)
Comment (mpbraendle) [reply CustomLocaleHandler.inc.php]: Anything that makes it easier to obtain the path is welcome - I'm just starting to dive into OJS programming. Is this function also available in OMP?

Comment (mpbraendle) [reply CustomLocaleHandler.inc.php]: OK, sorry - I have found it in OMP, too. Will implement it.

Comment (mpbraendle) [reply CustomLocaleHandler.inc.php]: See https://github.com/mpbraendle/customLocale/commit/a607c97e06f9cccadc93def056fb8763632c4a67

**Comment by asmecher** [comment]
@mpbraendle, please have a look at https://github.com/asmecher/customLocale/ -- this is preparatory work for the plugin to be released in a form compatible with OJS 3.2, but it should cover all of this as well.

**Comment by mpbraendle** [comment]
@asmecher - will do, but is OJS 3.2 required to test it? Looks like that in the new version all phrase files have a new format.

**Comment by asmecher** [comment]
Yes, that PR will require OJS 3.2 -- we're releasing that in about a month.

**Comment by asmecher** [comment]
@mpbraendle, just dusting off this old PR; have you explored the version of this plugin that's available in the Plugin Gallery (for new OJS/OMP/OPS releases)? Is this still relevant?

**Comment by asmecher** [comment]
@mpbraendle, just dusting off this old PR; have you explored the version of this plugin that's available in the Plugin Gallery (for new OJS/OMP/OPS releases)? Is this still relevant?

**Comment by asmecher** [comment]
@mpbraendle, just dusting off this old PR; have you explored the version of this plugin that's available in the Plugin Gallery (for new OJS/OMP/OPS releases)? Is this still relevant?

**Comment by mpbraendle** [comment]
@asmecher Just upgraded yesterday a single-journal installation to OJS 3.2.1 - this went smooth without any hassles. However, customLocale is not required there. The multijournal platform where some journals use customLocale will be upgraded early August - and we will test the plugin then.
